models/src/main/scala/coop/rchain/models/rholang/sort/ScoreTree.scala

```diff
@@ -22,25 +25,55 @@ trait ScoreTree {
 
   case class Node[T](children: Seq[Tree[T]]) extends Tree[T]
 
-  case class ScoreAtom(value: Either[Int, String]) {
-    def compare(that: ScoreAtom): Int = (this.value, that.value) match {
-      case (Left(i1), Left(i2))   => i1.compare(i2)
-      case (Left(_), Right(_))    => -1
-      case (Right(_), Left(_))    => 1
-      case (Right(s1), Right(s2)) => s1.compare(s2)
+  sealed trait TaggedAtom
+  case class IntAtom(i: Int)          extends TaggedAtom
+  case class StringAtom(s: String)    extends TaggedAtom
+  case class BytesAtom(b: ByteString) extends TaggedAtom
+
+  case class ScoreAtom(value: TaggedAtom) {
+    def bsCompare(b1: ByteString, b2: ByteString) = {
```

Comment:
We might think about eventually moving this to be part of an `implicit class RichByteString(...)` in `shared`

Reply:
I would approve that PR